### PR TITLE
Feature/267/not use max synchronous size new shmem adjustment

### PR DESCRIPTION
**Pull request by fercsrti77**
This pull request is coming from the issue #XX. When the pull request is merged into "master" the issue will be [resolved, closed, fixed] #XX

### CODE REVIEW

File: srcCpp/Infrastructure_pro.cxx [hunk 1]
```diff
@@ -580,10 +581,14 @@ bool configureShmemTransport(
     RTIBool success = RTI_FALSE;
     int retcode;
     int key = rand();
-    int minBufferSize = parentMsgSizeMax;
+    int minBufferSize = 1048576;
     int step = 1048576; // 1MB
     int maxBufferSize = (std::max)(60817408 /* 58MB */, parentMsgSizeMax);
 
+    if (!messageSizeMaxSet) {
```
Comment (javmorcas) [inline srcCpp/Infrastructure_pro.cxx]: What does this mean?

File: srcCpp/Infrastructure_pro.cxx [hunk 2]
```diff
@@ -537,9 +537,12 @@ bool configureShmemTransport(
...
-    } else if (qos.transport_builtin.mask == DDS_TRANSPORTBUILTIN_SHMEM) {
-        if ((datalen + MESSAGE_OVERHEAD_BYTES) > DEFAULT_MESSAGE_SIZE_MAX) {
-            parentMsgSizeMax = datalen + MESSAGE_OVERHEAD_BYTES;
+    } else {
+        if (qos.transport_builtin.mask == DDS_TRANSPORTBUILTIN_SHMEM) {
+            if ((datalen + MESSAGE_OVERHEAD_BYTES) > parentMsgSizeMax) {
+                parentMsgSizeMax = datalen + MESSAGE_OVERHEAD_BYTES;
+                transport.minimumMessageSizeMax = parentMsgSizeMax;
+            }
         }
```
Comment (javmorcas) [inline srcCpp/Infrastructure_pro.cxx]: ok. I agree with this part so far. Just to clarify though:

If you are setting the property for SHMEM but it is SHMEM + something, then you use the value of the property. If you are NOT setting the property and it is SHMEM + something, then you use whatever is the minimum for the max_size across all transports.

That kind of makes sense, but it is also a weird scenario.

File: srcCpp/Infrastructure_pro.cxx [hunk 3]
```diff
@@ -580,10 +581,14 @@ bool configureShmemTransport(
     RTIBool success = RTI_FALSE;
     int retcode;
     int key = rand();
-    int minBufferSize = parentMsgSizeMax;
+    int minBufferSize = 1048576;
     int step = 1048576; // 1MB
     int maxBufferSize = (std::max)(60817408 /* 58MB */, parentMsgSizeMax);
 
+    if (!messageSizeMaxSet) {
+
```
Comment (javmorcas) [inline srcCpp/Infrastructure_pro.cxx]: I am guessing this is in the wrong place, right? We are missing after the do-while a check with something like:

If size greater than 64k and I couldn't reserve memory for doing synchronous, default all the settings and try enabling asynchronous. Right?

File: srcCpp/Infrastructure_pro.cxx (hunk 2 above)
Comment (fercsrti77) [reply srcCpp/Infrastructure_pro.cxx]: If you set the property "dds.transport.shmem.builtin.parent.message_size_max" for SHMEM we always use the property. 

If you don't set the property we only set it if the only transport is SHMEM

File: srcCpp/Infrastructure_pro.cxx (hunk 3 above)
Comment (fercsrti77) [reply srcCpp/Infrastructure_pro.cxx]: In fact, that if had to be removed all together. Below we will adjust the transport size if messageSizeMaxSet is not true

File: srcCpp/Infrastructure_pro.cxx (hunk 2 above)
Comment (javmorcas) [reply srcCpp/Infrastructure_pro.cxx]: That makes sense to me.

**Pull request merged**